### PR TITLE
Add IResolver for UoW floating

### PR DESCRIPTION
**Pull request by WilliamBZA**
Fixes https://github.com/Particular/NServiceBus/issues/4744

There is some weirdness around the autofac `IContainerContext` where they don't want you to store a reference to the current `IContainerContext` and instead want you to use the current one to request another one....

@timbussmann solved the problem, I moved his test to a better location and added code comments.

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/Core/UnitOfWork/When_using_a_func_for_UOW_scope.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    {
+                        await bus.SendLocal(new Endpoint.MyMessage());
+                        await bus.SendLocal(new Endpoint.MyMessage());
+                        await bus.SendLocal(new Endpoint.MyMessage());
+                    });
+                })
+                .Done(c => c.CalledCount >= 3)
+                .Run(TimeSpan.FromSeconds(15));
+
+            Assert.AreEqual(3, UnitOrWorkInjected.InstanceCount);
```
Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Core/UnitOfWork/When_using_a_func_for_UOW_scope.cs]: given the test name, shouldn't there actually be multiple handlers on the same message (same uow) to resolve the same dependency?

Comment (WilliamBZA) [inline src/NServiceBus.AcceptanceTests/Core/UnitOfWork/When_using_a_func_for_UOW_scope.cs]: I don't agree, the name of the test matches with sending 3 separate messages which should result in 3 UOW scopes and therefore 3 separate instances.

Multiple handlers per message already resulted in the same instance anyway

Comment (timbussmann) [inline src/NServiceBus.AcceptanceTests/Core/UnitOfWork/When_using_a_func_for_UOW_scope.cs]: then that test would also pass using a transient scope

**Comment by WilliamBZA** [comment]
Closing this as it has been superseded by https://github.com/Particular/NServiceBus/pull/4832

**Pull request closed without merging**